api/webhook/taskcluster.go

```diff
@@ -88,8 +88,34 @@ type statusEventPayload struct {
 	Branches  []branchInfo `json:"branches"`
 }
 
+// GetBranch returns the first branch that has the revision of the commit
+// as its HEAD state.
+func (s statusEventPayload) GetBranch() string {
```

Comment:
Using this to decide whether a status update comes from the master branch might be problematic.

IIUC, the `branches` field contains a list of branches that contain this commit, each one with its *current* head. When a Taskcluster run finishes for a commit and posts the "success" status, that commit is very likely no longer the head of master.

I think we need two methods, like `IsOnMaster` and `HeadingBranches`, and only the latter looks at the SHA and will be used in the PR case.